mne/time_frequency/spectrum.py

```diff
@@ -0,0 +1,1186 @@
+                f'frequency of the data ({0.5 * inst.info["sfreq"]} Hz).')
+        _check_option('method', method, ('welch', 'multitaper'))
+
+        # triage method and kwargs. partial() doesn't check validity of kwargs,
+        # so we do it manually to save compute time if any are invalid.
+        psd_funcs = dict(welch=psd_array_welch,
+                         multitaper=psd_array_multitaper)
+        invalid_ix = np.in1d(list(method_kw),
+                             list(signature(psd_funcs[method]).parameters),
+                             invert=True)
+        if invalid_ix.any():
+            invalid_kw = np.array(list(method_kw))[invalid_ix].tolist()
+            s = _pl(invalid_kw)
+            raise TypeError(
+                f'Got unexpected keyword argument{s} {", ".join(invalid_kw)} '
+                f'for PSD method "{method}".')
+        self._psd_func = partial(psd_funcs[method], **method_kw)
+
+        # apply proj if desired
+        if proj:
+            inst = inst.copy().apply_proj()
+
+        # prep times and picks
+        self._time_mask = _time_mask(inst.times, tmin, tmax, sfreq=self.sfreq)
+        self._picks = _picks_to_idx(inst.info, picks, 'data',
+                                    with_ref_meg=False)
+
+        # add the info object. bads and non-data channels were dropped by
+        # _picks_to_idx() so we update the info accordingly:
+        self.info = pick_info(inst.info, sel=self._picks, copy=True)
+
+        # assign some attributes
+        self.preload = True  # needed for __getitem__, doesn't mean anything
+        self._method = method
+        self._inst_type = type(inst)
+        # self._dims may also get updated by child classes
+        self._dims = ('channel', 'freq',)
+        if method_kw.get('average', '') in (None, False):
+            self._dims += ('segment',)
+        if method_kw.get('output', '') == 'complex':
+            self._dims = self._dims[:-1] + ('taper',) + self._dims[-1:]
+        # record data type (for repr and html_repr)
+        self._data_type = ('Fourier Coefficients' if 'taper' in self._dims
+                           else 'Power Spectrum')
+
+    def __deepcopy__(self, memodict):
+        """Make a deepcopy."""
+        cls = self.__class__
+        result = cls.__new__(cls)
+        for k, v in self.__dict__.items():
+            v = deepcopy(v, memodict)
+            result.__dict__[k] = v
+        return result
+
+    def __eq__(self, other):
+        """Test equivalence of two Spectrum instances."""
+        # there is one attr (_inst_type) that object_diff can't handle
+        if self._inst_type != other._inst_type:
+            return False
+        _inst_type = self._inst_type
+        del self._inst_type
+        del other._inst_type
+        try:
+            same = object_diff(vars(self), vars(other)) == ''
+        finally:
+            self._inst_type = _inst_type
+            other._inst_type = _inst_type
```

Comment:
Maybe a better approach would be to make `object_diff` handle classes? I think in principle something like this could work:
```
if inspect.isclass(a):
    if not inspect.isclass(b):
        same = False
    else:
        same = (a == b)
```